lib/spack/spack/util/remote_file_cache.py

```diff
@@ -81,12 +81,12 @@ def local_path(raw_path: str, sha256: str, make_dest: Optional[Callable[[], str]
 
     # Allow paths (and URLs) to contain spack config/environment variables,
     # etc.
+    path = canonicalize_path(raw_path)
+
     win_path = pathlib.PureWindowsPath(raw_path)
```

Comment:
Given the fix at line 84, this should be passing `path`, not `raw_path`.

Reply:
That's to support the case that I hadn't handled properly yesterday; namely, that of a relative path that is made absolute using the current working directory since the `os` call adds the Windows drive.